assets/javascript/apps/pipeline/nodes/widgets.tsx

```diff
@@ -846,8 +849,80 @@ export function HistoryTypeWidget(props: WidgetParams) {
         <small className="text-red-500">{historyNameError}</small>
       </div>
     </>
-  )
-    ;
+  );
+}
+
+export function HistoryModeWidget(props: WidgetParams) {
+  const options = getSelectOptions(props.schema);
+  const userMaxTokenLimit = concatenate(props.nodeParams["user_max_token_limit"]);
+  const maxHistoryLength = concatenate(props.nodeParams["max_history_length"]);
+  const initialHistoryMode = concatenate(props.nodeParams["history_mode"]);
+  const [historyMode, setHistoryMode] = useState(initialHistoryMode || "Summarize");
+
+  const historyModeHelpTexts: Record<string, string> = {
+    summarize:"If the token count exceeds the limit, older messages will be summarized while keeping the last few messages intact.",
+    truncate_tokens:"If the token count exceeds the limit, older messages will be removed until the token count is below the limit.",
+    max_history_length:"The chat history will always be truncated to the last N messages.",
+  };
+
+  return (
+    <>
+      <div className="flex join">
+        <InputField label="History Mode">
+          <select
+            className="select select-bordered join-item"
+            name="history_mode"
+            onChange={(e) => {
+              setHistoryMode(e.target.value);
+              props.updateParamValue(e);
+            }}
+            value={historyMode}
+          >
+            {options.map((option) => (
+              <option key={option.value} value={option.value}>
+                {option.label}
+              </option>
+            ))}
+          </select>
+          <br />
+          <br />
+          <div>{historyModeHelpTexts[historyMode]}</div>
```

Comment:
this spacing should be done with a margin:
```suggestion
          <div className="mt-4">{historyModeHelpTexts[historyMode]}</div>
```